src/main/java/net/mcreator/ui/modgui/VillagerProfessionGUI.java

```diff
@@ -0,0 +1,212 @@
+
+		subpanel.add(HelpUtils.wrapWithHelpButton(this.withEntry("villagerprofession/hat"),
+				L10N.label("elementgui.villager_profession.hat")));
+		subpanel.add(hat);
+
+		JButton importProfessionTexture = new JButton(UIRES.get("18px.add"));
+		importProfessionTexture.setToolTipText(L10N.t("elementgui.villager_profession.import_profession_texture"));
+		importProfessionTexture.setOpaque(false);
+		importProfessionTexture.addActionListener(e -> {
+			TextureImportDialogs.importMultipleTextures(mcreator, TextureType.ENTITY);
+			professionTextureFile.removeAllItems();
+			professionTextureFile.addItem("");
+			mcreator.getFolderManager().getTexturesList(TextureType.ENTITY)
+					.forEach(el -> professionTextureFile.addItem(el.getName()));
+		});
+
+		subpanel.add(HelpUtils.wrapWithHelpButton(this.withEntry("villagerprofession/profession_texture"),
+				L10N.label("elementgui.villager_profession.profession_texture")));
+		subpanel.add(PanelUtils.centerAndEastElement(professionTextureFile, importProfessionTexture));
+
+		JButton importZombifiedProfessionTexture = new JButton(UIRES.get("18px.add"));
+		importZombifiedProfessionTexture.setToolTipText(
+				L10N.t("elementgui.villager_profession.import_zombified_profession_texture"));
+		importZombifiedProfessionTexture.setOpaque(false);
+		importZombifiedProfessionTexture.addActionListener(e -> {
+			TextureImportDialogs.importMultipleTextures(mcreator, TextureType.ENTITY);
+			zombifiedProfessionTextureFile.removeAllItems();
+			zombifiedProfessionTextureFile.addItem("");
+			mcreator.getFolderManager().getTexturesList(TextureType.ENTITY)
+					.forEach(el -> zombifiedProfessionTextureFile.addItem(el.getName()));
+		});
+
+		subpanel.add(HelpUtils.wrapWithHelpButton(this.withEntry("villagerprofession/zombified_profession_texture"),
+				L10N.label("elementgui.villager_profession.zombified_profession_texture")));
+		subpanel.add(PanelUtils.centerAndEastElement(zombifiedProfessionTextureFile, importZombifiedProfessionTexture));
+
+		page1group.addValidationElement(displayName);
+		page1group.addValidationElement(pointOfInterest);
+		page1group.addValidationElement(actionSound.getVTextField());
+		page1group.addValidationElement(professionTextureFile);
+		page1group.addValidationElement(zombifiedProfessionTextureFile);
+
+		displayName.setValidator(new TextFieldValidator(displayName,
+				L10N.t("elementgui.villager_profession.profession_needs_display_name")));
+		displayName.enableRealtimeValidation();
+		pointOfInterest.setValidator(
+				new UniqueNameValidator(L10N.t("elementgui.villager_profession.profession_block_validator"),
+						() -> pointOfInterest.getBlock().getUnmappedValue(),
+						() -> ElementUtil.loadAllPOIBlocks(mcreator.getWorkspace()).stream()
+								.map(MItemBlock::getUnmappedValue),
+						ElementUtil.loadBlocks(mcreator.getWorkspace()).stream().filter(MCItem::isPOI)
+								.map(DataListEntry::getName).toList(), null).setIsPresentOnList(isEditingMode()));
```

Comment:
After #3640 is merged:
```suggestion
								.map(DataListEntry::getName).toList(), null).setIsPresentOnList(this::isEditingMode));
```

Reply:
>After #3640 is merged:

Or Max094Reikeb#266 as an alternative.

Reply:
Done!